File: HRDashboard/HRDashboard/Models/Project.cs
namespace HRDashboard.Models;

public class Project
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string EmbedUrl { get; set; } = string.Empty;

    public Guid? CategoryId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public Category? Category { get; set; }
}

File: frontend/src/types/project.ts
export interface ProjectResponse {
  id: string;
  name: string;
  description: string | null;
  embedUrl: string;
  categoryId: string | null;
  categoryName: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProject {
  name: string;
  description: string | null;
  embedUrl: string;
  categoryId: string | null;
}

export type UpdateProject = CreateProject;